enhancements/profile-versioning.md

```diff
@@ -0,0 +1,413 @@
+3. As a cluster administrator, I want to be able to opt into having the latest
+   benchmark scanning my cluster at all times.
+
+### API Extensions
+
+This enhancement focuses solely on adding a version to the `Profile` CRD,
+either through a formal attribute, label, or annotation. No other resources
+will be modified as a result of this implementation.
+
+The implementation should be backwards compatible so that if a profile does not
+have a version in the data stream, the resulting `Profile` resource will have
+an empty value for `version`.
+
+### Version Evaluation
+
+Profile version semantics should follow [semantic versioning
+v2.0.0](https://semver.org/spec/v2.0.0.html), which is the latest supported
+version of SemVer at the time of this writing. A summary of the differences
+between version 1.0.0 and 2.0.0 are available in a [GitHub
+issue](https://github.com/semver/semver/issues/231).
+
+This would allow us to use the following strings to represent profile versions:
+
+* `1.2.0` being the most basic form of versions we see in compliance benchmarks
+* `1.2.0-1` includes flexibility to denote differences across the same profile version
+* `1.2.0-1+e076b1d7ac` include the content of the profile as a SHA
+
+Initially, we may choose to only implement simple versions (e.g., 1.2.0, or
+4.0.0). Depending on user feedback, we have the option to include support for
+more complex version strings that use pre-release or build metadata within the
+semantic version specification to denote changes to a profile between operator
+releases. If we do decide to support pre-release or build metadata as part of
+the version string, we should ensure version string evaluation considers
+`1.2.0` older than `1.2.0-1` or `1.2.0-2+e076b1d7ac`. This detail will be
+covered in a future enhancement if we decide to implement granular profile
+versions, as noted in the non-goals section above.
+
+Note that we can't enforce semantic versioning because the version is
+ultimately set by the authoring body of the standard.
+
+### Implementation Details/Notes/Constraints [optional]
+
+There are two considerations for how versions interact with profiles.
+
+The first is if there is only one profile, and it has a single version. For
+example, supporting the CIS OpenShift 1.4.0 benchmark. The second is if there
+is a single profile, but multiple versions. For example, PCI-DSS version 3.2.1
+and version 4.0.0.
+
+#### Single Profile & Single Version
+
+This case can be implemented in the following steps:
+
+1. Associate a version to the profile in
+   [ComplianceAsCode/content](https://github.com/ComplianceAsCode/content)
+   using the `metadata.version` attribute of the profile
+2. Update the operator's profile parser to parse the version from the data
+   stream, and set the appropriate attribute on the `Profile` custom resource
+
+This will look similar to the following after the profiles are parsed by the
+operator on start-up:
+
+```console
+$ oc get profiles.compliance
```

Comment:
if no special setting, `AGE` field is not available for this command.
```
$ oc get profile.compliance
NAME                             VERSION
ocp4-cis                         1.4.0
ocp4-cis-node                    1.4.0
ocp4-e8                          
ocp4-high                        Revision 4
ocp4-high-node                   Revision 4
ocp4-moderate                    Revision 4
ocp4-moderate-node               Revision 4
ocp4-nerc-cip                    
ocp4-nerc-cip-node               
ocp4-pci-dss                     3.2.1
ocp4-pci-dss-node                3.2.1
ocp4-stig                        V1R1
ocp4-stig-node                   V1R1
rhcos4-anssi-bp28-enhanced       
rhcos4-anssi-bp28-high           
rhcos4-anssi-bp28-intermediary   
rhcos4-anssi-bp28-minimal        
rhcos4-e8                        
rhcos4-high                      Revision 4
rhcos4-moderate                  Revision 4
rhcos4-nerc-cip                  
rhcos4-stig                      V1R1

```

Reply:
Ack - working on adding this into the output for Profiles so it's consistent with what's done today.

Reply:
Something like this is what you'd like to see, right?

```
 $ oc get profiles.compliance
NAME                             AGE   VERSION
ocp4-cis                         21s   1.4.0
ocp4-cis-node                    22s   1.4.0
ocp4-e8                          21s
ocp4-high                        21s   Revision 4
ocp4-high-node                   21s   Revision 4
ocp4-moderate                    21s   Revision 4
ocp4-moderate-node               21s   Revision 4
ocp4-nerc-cip                    21s
ocp4-nerc-cip-node               21s
ocp4-pci-dss                     21s   3.2.1
ocp4-pci-dss-node                21s   3.2.1
ocp4-stig                        21s   V1R1
ocp4-stig-node                   21s   V1R1
rhcos4-anssi-bp28-enhanced       12s
rhcos4-anssi-bp28-high           11s
rhcos4-anssi-bp28-intermediary   11s
rhcos4-anssi-bp28-minimal        11s
rhcos4-e8                        11s
rhcos4-high                      11s   Revision 4
rhcos4-moderate                  11s   Revision 4
rhcos4-nerc-cip                  11s
rhcos4-stig                      11s   V1R1
```

Reply:
Yes, age field is available for the latest version of https://github.com/ComplianceAsCode/compliance-operator/pull/425. A minor issue about some profiles have a empty value for the version field, such as nerc-cip profile, is it expected? Thanks.
$ oc get profile.compliance
NAME AGE VERSION
ocp4-cis 2m20s 1.4.0
ocp4-cis-node 2m20s 1.4.0
ocp4-e8 2m20s
ocp4-high 2m20s Revision 4
ocp4-high-node 2m20s Revision 4
ocp4-moderate 2m20s Revision 4
ocp4-moderate-node 2m20s Revision 4
ocp4-nerc-cip 2m20s
ocp4-nerc-cip-node 2m20s
ocp4-pci-dss 2m20s 3.2.1
ocp4-pci-dss-node 2m20s 3.2.1
ocp4-stig 2m20s V1R1
ocp4-stig-node 2m20s V1R1
rhcos4-anssi-bp28-enhanced 2m17s
rhcos4-anssi-bp28-high 2m16s
rhcos4-anssi-bp28-intermediary 2m16s
rhcos4-anssi-bp28-minimal 2m16s
rhcos4-e8 2m16s
rhcos4-high 2m16s Revision 4
rhcos4-moderate 2m16s Revision 4
rhcos4-nerc-cip 2m16s